documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,252 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'prometheus',
+        rules: [
+          {
+            alert: 'PrometheusBadConfig',
+            expr: |||
+              prometheus_config_last_reload_successful{%(prometheusSelector)s} == 0
```

Comment:
You should use max_over_time/min_over_time here, otherwise a failed scrape will reset the alert due to staleness. Similarly with any other gauge-based alerts.

Reply:
Ooof. That's a very good point. All nicely written up in https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 , but which I totally forgot about. This is probably a good opportunity to spread awareness. I'll rework all gauge-based alerts.